### PR TITLE
Allowing DataFrames where all columns are DoubleType to be a source for RowPartitionedMatrix

### PR DESCRIPTION
**Pull request by dputler**
One underlying question I have is what is the most efficient way of getting data out of the Row objects created by map. The current work line of code is: 
val matrixRDD = df.map(x => x.toSeq.toArray).map(y => y.map(z => z.asInstanceOf[Double])

### CODE REVIEW

File: src/main/scala/edu/berkeley/cs/amplab/mlmatrix/RowPartitionedMatrix.scala [hunk 1]
```diff
@@ -311,6 +314,80 @@ object RowPartitionedMatrix {
...
+        }
+        nRow += 1
+      }
+      Iterator(new DenseMatrix[Double](rows, cols, matData.toArray))
+    }
+    data
+  }
+
+  def dataFrameToMatrix(df: DataFrame): RDD[DenseMatrix[Double]] = {
+    val matrixRDD = df.map(x => x.toSeq.toArray).map(y => y.map(z => z.asInstanceOf[Double]))
```
Comment (shivaram) [inline src/main/scala/edu/berkeley/cs/amplab/mlmatrix/RowPartitionedMatrix.scala]: If I'm not wrong `matrixRDD` here is a `RDD[Array[Double]]` ? If so can we just call `arrayToMatrix(matrixRDD)` here ?

File: src/main/scala/edu/berkeley/cs/amplab/mlmatrix/RowPartitionedMatrix.scala [hunk 2]
```diff
@@ -311,6 +314,80 @@ object RowPartitionedMatrix {
...
+      dataFrameToMatrix(df, rowsPerPartition, cols).map(mat => RowPartition(mat)),
+        Some(rowsPerPartition.sum), Some(cols))
+  }
+
+  // Convert a DataFrame of all DoubleType columns to a RDD[DenseMatrix[Double]]
+  def dataFrameToMatrix(
+      df: DataFrame,
+      rowsPerPartition: Seq[Int],
+      cols: Int) = {
+    val matrixRDD = df.map(x => x.toSeq.toArray).map(y => y.map(z => z.asInstanceOf[Double]))
```
Comment (shivaram) [inline src/main/scala/edu/berkeley/cs/amplab/mlmatrix/RowPartitionedMatrix.scala]: Similar comment to below, can we reuse the arrayToMatrix function here ?

**Comment by shivaram** [comment]
Thanks @dputler for the PR. I think it looks pretty good and just had a question about reusing some of the existing methods.

BTW I think the toSeq.map(_ => _.toDouble) is the best way to go from a Row of doubles to Array[Double].

**Comment by shivaram** [comment]
Also is there a specific version of Spark version you are using ? Right now the ml-matrix default build is using 1.2.1 and I might upgrade it to 1.3.1 or 1.4.0

File: src/main/scala/edu/berkeley/cs/amplab/mlmatrix/RowPartitionedMatrix.scala (hunk 1 above)
Comment (dputler) [inline src/main/scala/edu/berkeley/cs/amplab/mlmatrix/RowPartitionedMatrix.scala]: Yes, I can make that change. I'm currently violating "don't repeat yourself".

File: src/main/scala/edu/berkeley/cs/amplab/mlmatrix/RowPartitionedMatrix.scala (hunk 2 above)
Comment (dputler) [inline src/main/scala/edu/berkeley/cs/amplab/mlmatrix/RowPartitionedMatrix.scala]: Similar response to the above. I'll make the change.

**Comment by dputler** [comment]
I've been working with a 1.4.0-SNAPSHOT, which is what we are using on the R side. Things should work with 1.3.1 out of the box, 1.2.1 may be a different story.

**Comment by shivaram** [comment]
Thanks @dputler -- Could you also update the default spark version to 1.3.1 in the build.sbt file ?[1] The default build otherwise fails as DataFrames were introduced in 1.3 

[1] https://github.com/amplab/ml-matrix/blob/master/build.sbt#L16

**Comment by dputler** [comment]
Will do.

**Comment by shivaram** [comment]
LGTM. Merging this

**Pull request merged**